get_alma_data_by_call_number.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import tomllib
+import requests
+import json
+import xmltodict
+
+
+def _get_args() -> argparse.Namespace:
+    """Returns the command-line arguments for this program.
+
+    :return: Parsed CLI arguments.
+    """
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--config_file",
+        help="Path to config file with Alma connection info",
+        required=True,
+    )
+
+    parser.add_argument(
+        "--call_number",
+        help="Call number of record to match between the DL Django app and Alma",
+        required=True,
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def _get_config(config_file_name: str) -> dict:
+    """Returns configuration for this program, loaded from TOML file.
+
+    :param str config_file_name: Filename for the TOML-format config file.
+    :return: A dict with the config info.
+    """
+
+    with open(config_file_name, "rb") as f:
+
+        config = tomllib.load(f)
+
+    return config
+
+
+def get_alma_data_by_call_number(sru_url: str, call_number: str) -> dict:
+    """Fetches Alma data for a given call number.
+
+    :param sru_url: Base URL for the Alma SRU API.
+    :param call_number: Call number to search in Alma.
+    :return: dictionary containing all data from original XML response.
+    """
+
+    alma_url_parameters = (
+        "?version=1.2&operation=searchRetrieve&recordSchema=marcxml"
+        "&query=alma.PermanentCallNumber="
+    )
+    full_sru_url = f"{sru_url}{alma_url_parameters}{call_number}"
+
+    response = requests.get(full_sru_url)
+
+    if response.status_code != 200:
+        raise Exception(f"Error fetching data from Alma: {response.text}")
+
+    # Convert XML response to a dictionary
+    xml_dict = xmltodict.parse(response.text)
+    return xml_dict
+
+
+def get_relevant_fields_alma(alma_data: dict) -> list:
+    """Extracts only the relevant fields from the Alma data.
+
+    :param alma_data: The full Alma data dictionary, as obtained from the SRU API,
+        assumed to contain only one record.
+    :return: A list of relevant fields extracted from the Alma data.
+    """
+    # SearchRetrieveResponse is the top-level key in the Alma SRU response
+    if "searchRetrieveResponse" not in alma_data:
+        return []
+
+    # Assuming there is only one record in the response, this is the expected structure
```

Comment:
How is the structure different if multiple records are returned?  Let's document this now while looking at the responses.  For possible examples of inventory numbers which are in multiple records, you can use the CSV generated by `get_ftva_holdings_report.py` to find values which occur multiple times.